homeassistant/components/media_player/__init__.py

```diff
@@ -928,6 +969,55 @@ async def websocket_handle_thumbnail(hass, connection, msg):
     )
 
 
+@websocket_api.async_response
+async def websocket_browse_media(hass, connection, msg):
+    """
+    Browse media available to the media_player entity.
+
+    To use, media_player integrations can implement MediaPlayerEntity.async_browse_media()
+    """
+    component = hass.data[DOMAIN]
+    player = component.get_entity(msg["entity_id"])
+
+    if player is None:
+        connection.send_message(
+            websocket_api.error_message(
+                msg["id"], "entity_not_found", "Entity not found"
+            )
+        )
+        return
+
+    if not player.supported_features & SUPPORT_BROWSE_MEDIA:
+        connection.send_message(
+            websocket_api.error_message(
+                msg["id"], "browse_failed", "Player does not support browsing media"
+            )
+        )
+        return
+
+    media_content_type = msg.get("media_content_type")
+    media_content_id = msg.get("media_content_id")
+
+    try:
+        payload = await player.async_browse_media(media_content_type, media_content_id)
+    except NotImplementedError:
+        connection.send_message(
+            websocket_api.error_message(
+                msg["id"],
+                "browse_failed",
+                "Integration does not support browsing media",
+            )
+        )
+        return
+    except HomeAssistantError as err:
+        connection.send_message(
+            websocket_api.error_message(msg["id"], "browse_failed", str(err))
```

Comment:
ERR_UNKNOWN_ERROR and also print stack trace. This might be the default behavior though if you don't catch it?